pages/visualizations/overview/active_drifting_contributors.py

```diff
@@ -158,34 +152,34 @@ def toggle_popover_4(n, is_open):
 @callback(
     Output("active_drifting_contributors", "figure"),
     Output("drifting_away_check_alert", "is_open"),
-    Output("active-drifting-contributors-timer", "n_intervals"),
     [
         Input("repo-choices", "data"),
-        Input("active-drifting-contributors-timer", "n_intervals"),
         Input("active-drifting-interval", "value"),
         Input("drifting_months", "value"),
         Input("away_months", "value"),
     ],
+    background=True,
 )
-def active_drifting_contributors(repolist, timer_pings, interval, drift_interval, away_interval):
-
-    logging.debug("ADC - PONG")
+def active_drifting_contributors(repolist, interval, drift_interval, away_interval):
 
     if drift_interval is None or away_interval is None:
         return dash.no_update, dash.no_update, dash.no_update
 
     if drift_interval > away_interval:
         return dash.no_update, True, dash.no_update
 
-    ready, results, graph_update, interval_update = handle_job_state(jm, ctq, repolist)
-    if not ready:
-        return graph_update, dash.no_update, interval_update
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=ctq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=ctq, repos=repolist)
 
-    logging.debug("ACTIVE_DRIFTING_CONTRIBUTOR_GROWTH_VIZ - START")
+    logging.debug(f"ACTIVE_DRIFTING_CONTRIBUTOR_GROWTH_VIZ - START")
     start = time.perf_counter()
 
-    # create dataframe from record data
-    df = pd.DataFrame(results)
+    if df.empty:
+        return nodata_graph, False
 
     # test if there is data
     if df.empty:
```

Comment:
if df.empty statements in there twice

Reply:
ah good catch, copy-paste error on my part